connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConnector.java

```diff
@@ -80,19 +86,14 @@ public void initialize(ConnectorConfig connectorConfig) {
                 SinkConnectorConfig.validate(config);
             }
 
-            connector.initialize(new ConnectorContext() {
-                @Override
-                public void requestTaskReconfiguration() {
-                    ctx.requestTaskReconfiguration();
-                }
-
-                @Override
-                public void raiseError(Exception e) {
-                    log.error("{} Connector raised an error", WorkerConnector.this, e);
-                    onFailure(e);
-                    ctx.raiseError(e);
-                }
-            });
+            if (isSinkConnector()) {
+                SinkConnectorConfig.validate(config);
+                connector.initialize(new WorkerSinkConnectorContext());
+            } else if (isSourceConnector()) {
+                connector.initialize(new WorkerSourceConnectorContext(offsetStorageReader));
+            } else {
+                connector.initialize(new WorkerConnectorContext());
```

Comment:
We don't allow a third option [elsewhere](https://github.com/apache/kafka/blob/trunk/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java#L1288-L1292) in the code, so probably we shouldn't here either.

Reply:
Good call. I'll update the code to throw an exception like in `Worker` when creating a source and sink task.